syne_tune/backend/sagemaker_backend/sagemaker_backend.py

```diff
@@ -135,10 +151,23 @@ def _all_trial_results(self, trial_ids: List[int]) -> List[TrialResult]:
 
         # overrides the status return by Sagemaker as the stopping decision may not have been propagated yet.
         for trial_res in res:
-            if trial_res.trial_id in self.paused_jobs:
+            trial_id = trial_res.trial_id
+            if trial_id in self.paused_jobs:
                 trial_res.status = Status.paused
-            if trial_res.trial_id in self.stopped_jobs:
+            if trial_id in self.stopped_jobs:
                 trial_res.status = Status.stopped
+            if trial_id in self._busy_trial_ids:
+                status = trial_res.status
+                if status not in BUSY_STATUS:
+                    self._busy_trial_ids.remove(trial_id)  # not busy anymore
+                    if trial_id in self._stopping_time:
+                        stop_time = time.time() - self._stopping_time[trial_id]
+                        logger.info(
+                            f"Stopping delay for trial_id {trial_id}: {stop_time:.2f} secs"
+                        )
+                        del self._stopping_time[trial_id]
+                elif status == Status.stopping and trial_id not in self._stopping_time:
+                    self._stopping_time[trial_id] = time.time()
```

Comment:
Yes, but that is just a "nice to have" functionality to show that stop delays are currently really long. So if delays are very short, I don't even want that message.

This is probably going to be removed, once SageMaker EASE fixes these stop delays.

Reply:
I see, we should add a comment about this in the logging info since now it suggest we know the number accurately. Maybe also add a comment in the code to indicate this.

Reply:
OK